src/lsptoolshost/profiling/profiling.ts

```diff
@@ -24,3 +30,271 @@ export function getProfilingEnvVars(outputChannel: LogOutputChannel): NodeJS.Pro
 
     return profilingEnvVars;
 }
+
+export function registerTraceCommand(
+    context: vscode.ExtensionContext,
+    languageServer: RoslynLanguageServer,
+    outputChannel: vscode.LogOutputChannel
+): void {
+    context.subscriptions.push(
+        vscode.commands.registerCommand('csharp.recordTrace', async () => {
+            await vscode.window.withProgress(
+                {
+                    location: vscode.ProgressLocation.Notification,
+                    title: 'dotnet-trace',
+                    cancellable: true,
+                },
+                async (progress, token) => {
+                    progress.report({
+                        message: vscode.l10n.t({
+                            message: 'Initializing dotnet-trace...',
+                            comment: 'dotnet-trace is a command name and should not be localized',
+                        }),
+                    });
+                    try {
+                        await executeDotNetTraceCommand(languageServer, progress, outputChannel, token);
+                    } catch (error) {
+                        const errorMessage = error instanceof Error ? error.message : String(error);
+                        showErrorMessageWithOptions(
+                            vscode,
+                            vscode.l10n.t({
+                                message: 'Failed to execute dotnet-trace command: {0}',
+                                args: [errorMessage],
+                                comment: 'dotnet-trace is a command name and should not be localized',
+                            }),
+                            { modal: true }
+                        );
+                    }
+                }
+            );
+        })
+    );
+}
+
+async function executeDotNetTraceCommand(
+    languageServer: RoslynLanguageServer,
+    progress: vscode.Progress<{
+        message?: string;
+        increment?: number;
+    }>,
+    outputChannel: vscode.LogOutputChannel,
+    cancellationToken: vscode.CancellationToken
+): Promise<void> {
+    const processId = languageServer.processId;
+
+    if (!processId) {
```

Comment:
Should this launch the process in this case? That way there's a good chance the trace might start earlier if the user needs to trace something right away?

Reply:
1.  The command is only registered after the server starts (to get this processId).  So its not expected to be missing
2.  The most correct way to do a startup trace is to invoke the language server process via dotnet-trace.  But that requires a bit more refactoring that I didn't have time for my timebox.